### PR TITLE
Organ damage

### PR DESCRIPTION
**Pull request by jroinc**
## About The Pull Request

Organs now have a percentage chance to heal 0.2 damage per tick. This chance starts at 20%, gains +20% if peri is present in the bloodstream, +10 if inaprov is present in the bloodstream, and loses 2% for every point of damage taken.

Thus if you have peri and inaprov in your bloodstream, you could theoretically heal from 25 (24.5 actually) organ damage (at a 1% chance per tick for 0.2 heal, granted, but it *could* happen), and will more likely likely be able to gradually heal from ~10-15 points of damage, given 10-15 minutes or so. That's the positive side.

The negative side is if you have a broken organ (30+ damage), it now has a chance to get worse. If you have a broken organ, and you do *not* have peri in you, you roll a chance for it to worsen. The chance starts at 60%, gains 2% per point of damage, and loses 40% if you have inaprov in your bloodstream. If the percentage chance passes, gain 0.2 damage.

Additionally, some organ damages were uncapped when broken. In addition to the base 1.3 bloodloss for a broken heart, there's an additional 0.03 bloodloss per point of damage. Lung damage deals 0.1 losebreath per point above 30, set to begin at 41 damage due to the way losebreath works, so there's some grace damage there. Liver and kidneys were already uncapped, so they're good.

Brain damage was *not* uncapped, but the minimum threshold for a force-rest or item drop was raised from *any amount* to five. You're welcome.

## Why It's Good For The Game

Better organ damage mechanics. If you aren't badly hurt, it'll go away with time and maybe a little medicine. If you *are* badly hurt and can't get to treatment, it'll kill you off quicker without you lingering forever in crit/nearcrit.

also now getting scratched once in the head won't make you drop your shit and collapse over and over again forever. Unless it actually *was* that bad.

## Changelog
:cl:
balance: rebalanced organ damage. Small amounts (10 or less) will autoheal over time, and peri and inaprov can help with this process. Organs that are broken (30+ damage) and not stabilized with peri will get worse over time, though inaprov can mitigate the damage partially.
/:cl:

### CODE REVIEW

**Comment by Lumipharon** [comment]
The brain damage thing also seemed a bit odd to me (probably realistic, but not ideal for a game).

**Comment by jroinc** [comment]
> 
> 
> The brain damage thing also seemed a bit odd to me (probably realistic, but not ideal for a game).

Honestly? I was minded to *up* the chance as it's only 1.5% of happening, but that's for a later time.

**Comment by jroinc** [comment]
Tested, *partially* functions as intended. Peri and inaprov don't seem to boost healrate as intended, will diagnose tomorrow, unless someone else beats me to it. Probably a var issue with my detection method.

**Comment by bowlofcereal** [comment]
Will there be a way to stop combat robots from receiving brain damage in the future? They have literally no way to fix this outside of aheals or the autosurgeon.

**Comment by ThelastArchiver** [comment]
might wanna look at #8773 inaprov would insta bring organs down to 15 damage and the rest could then just heal. making medevacing marines because of organ damage unnessecary except  for bloodloss and hugged marines and delimbed ones.

**Comment by jroinc** [comment]
> 
> 
> Will there be a way to stop combat robots from receiving brain damage in the future? They have literally no way to fix this outside of aheals or the autosurgeon.

Not from this.

**Comment by jroinc** [comment]
> 
> 
> might wanna look at #8773 inaprov would insta bring organs down to 15 damage and the rest could then just heal. making medevacing marines because of organ damage unnessecary except for bloodloss and hugged marines and delimbed ones.

It's already unneeded. No one gets medevac'd for organ damage, far as I can tell.

File: code/modules/organs/organ_internal.dm [hunk 1]
```diff
@@ -108,6 +108,23 @@
 			if (prob(3))	//about once every 30 seconds
 				take_damage(1, prob(30))
 
+	//Process natural and assisted healing
+	var/peridaxon = owner.reagents.has_reagent(/datum/reagent/medicine/peridaxon)
+	var/inaprovaline = owner.reagents.has_reagent(/datum/reagent/medicine/inaprovaline)
+	var/organ_heal_chance = 0
```
Comment (hyper2snyper) [inline code/modules/organs/organ_internal.dm:114]: this is used once, so you can just insert the assignment into the if(prob

File: code/modules/organs/organ_internal.dm [hunk 2]
```diff
@@ -108,6 +108,23 @@
...
+	var/organ_hurt_chance = 0
+	organ_heal_chance = 20 + (20*peridaxon) + (10*inaprovaline) - (2*damage)
```
Comment (hyper2snyper) [inline code/modules/organs/organ_internal.dm:116]: same as above

File: code/modules/organs/organ_internal.dm [hunk 3]
```diff
@@ -108,6 +108,23 @@
 			if (prob(3))	//about once every 30 seconds
 				take_damage(1, prob(30))
 
+	//Process natural and assisted healing
+	var/peridaxon = owner.reagents.has_reagent(/datum/reagent/medicine/peridaxon)
+	var/inaprovaline = owner.reagents.has_reagent(/datum/reagent/medicine/inaprovaline)
+	var/organ_heal_chance = 0
+	var/organ_hurt_chance = 0
```
Comment (hyper2snyper) [inline code/modules/organs/organ_internal.dm:115]: same as above. you dont need.
var/organ_hurt
organ_hurt = number
if(organ_hurt)

instead you can just do

if(number)

**Review by hyper2snyper — COMMENTED** [review]
good

**Comment by jroinc** [comment]
Did the things, however it still doesn't... actually work in testing. The base organ heal rate below 10 points is good, but peri and inaprov appear to have no effect.

File: code/modules/organs/organ_internal.dm [hunk 4]
```diff
@@ -108,6 +108,16 @@
 			if (prob(3))	//about once every 30 seconds
 				take_damage(1, prob(30))
 
+	//Process natural and assisted healing
+	if(prob(20 + (20*(owner.reagents.has_reagent(/datum/reagent/medicine/peridaxon, 0.05))) + (10*(owner.reagents.has_reagent(/datum/reagent/medicine/inaprovaline, 0.05))) - (2*damage)))
+		heal_organ_damage(0.2) //20% base chance, +20% if inaprov, +10% if peridaxon, -2% per point of damage
+	//And process natural hurting
+	if(owner.reagents.get_reagent_amount(/datum/reagent/medicine/peridaxon) >= 0.05)
+		return
+	if(is_broken())
```
Comment (hyper2snyper) [inline code/modules/organs/organ_internal.dm:117]: This can be one if statement.

**Review by BraveMole — CHANGES_REQUESTED** [review]
what hyper said, rest lgtm

**Comment by jroinc** [comment]
> what hyper said, rest lgtm

yes but it doesn't work. It doesn't recognize the presence of peri or inaprov.

File: code/modules/organs/organ_internal.dm [hunk 5]
```diff
@@ -108,6 +108,16 @@
...
+		if(prob((2*damage) - (40*(owner.reagents.has_reagent(/datum/reagent/medicine/inaprovaline, 0.05)))))
```
Comment (BraveMole) [inline code/modules/organs/organ_internal.dm:118]:
```suggestion
	if(is_broken() && (prob((2*damage) - (40*(owner.reagents.has_reagent(/datum/reagent/medicine/inaprovaline, 0.05))))))
```

**Review by BraveMole — CHANGES_REQUESTED** [review]
your code doesn't compile, identation issues and one too many ( i think

**Comment by BraveMole** [comment]
stale

**Pull request closed without merging**